### DISCUSSION TITLE
Facing issue in go unit test

### DISCUSSION THREAD

**Discussion by novertia** [milvus-io/milvus#36737] [Q&A and General discussion]
I was running go unit test following this document [https://github.com/milvus-io/milvus/blob/master/DEVELOPMENT.md]. Getting the following issue 

<img width="1262" alt="image" src="https://github.com/user-attachments/assets/a58c6f5e-0bb1-42fe-8b3f-7731565dfbab">
it look like a test are trying to open rocksdb without the lock file being cleared.

Not sure if I am doing something wrong.

System I am running on 
wsl2 - Ubuntu 22.04 LTS
RAM - 24gb allocated
Cores - 8 
SSD free space 112 Gb

**Comment by yhmo** [comment]
Try deleting the "/tmp/milvus" directory and run again

**Comment by novertia** [comment]
tried that but this time tested the components seperately, like running the proxy unit test then data node etc. Other components did not see this issue but in data node even when /tmp/milvus is not there still getting this. When data node test starts this directory is being created then after sometime I am getting the issue. From what I could understand from the behaviour it seems like 1 test case in data node open rocks db with write option but never closes it and another tries to open it with as write again.

**Reply by yhmo** [reply]
Follow these steps:

Replace all the "/var/lib/milvus" to be "/tmp/milvus" in the configs/milvus.yaml
```
cd cd deployments/docker/dev
sudo rm -rf ./volumes
docker compose up -d
cd -
make test-go
```
![Screenshot from 2024-10-11 16-37-06](https://github.com/user-attachments/assets/24d8a316-e67b-46d5-925e-7794aac0f9ea)


If you want to run case respectively, you can:
`go test -v "./internal/util/bloomfilter..." -failfast -test.run TestPerformance_MultiBF`

**Comment by novertia** [comment]
milvus commit I am running unit test on : 3685edb264c2f4de691d35a719555c05968d40d8 

Hi, 

I tested out using the steps u mentioned. proxy test cases there is no issues. However during running of datanode test cases I am facing the same issue. I ran the datanode test cases only after clearing out tmp and volume and the issue persist.
